Spigot-API-Patches/0006-Adventure-Events.patch

```diff
@@ -0,0 +1,1062 @@
+         this(player, drops, droppedExp, 0, deathMessage);
+     }
+ 
++    @Deprecated // Paper
+     public PlayerDeathEvent(@NotNull final Player player, @NotNull final List<ItemStack> drops, final int droppedExp, final int newExp, @Nullable final String deathMessage) {
+         this(player, drops, droppedExp, newExp, 0, 0, deathMessage);
+     }
+ 
++    @Deprecated // Paper
+     public PlayerDeathEvent(@NotNull final Player player, @NotNull final List<ItemStack> drops, final int droppedExp, final int newExp, final int newTotalExp, final int newLevel, @Nullable final String deathMessage) {
+         super(player, drops, droppedExp);
+         this.newExp = newExp;
+         this.newTotalExp = newTotalExp;
+         this.newLevel = newLevel;
+-        this.deathMessage = deathMessage;
++        this.deathMessage = deathMessage != null ? net.kyori.adventure.text.serializer.legacy.LegacyComponentSerializer.legacySection().deserialize(deathMessage) : net.kyori.adventure.text.Component.empty(); // Paper
+     }
+ 
+     @NotNull
+@@ -39,12 +59,13 @@ public class PlayerDeathEvent extends EntityDeathEvent {
+         return (Player) entity;
+     }
+ 
++    // Paper start
+     /**
+      * Set the death message that will appear to everyone on the server.
+      *
+      * @param deathMessage Message to appear to other players on the server.
+      */
+-    public void setDeathMessage(@Nullable String deathMessage) {
++    public void deathMessage(@Nullable net.kyori.adventure.text.Component deathMessage) {
+         this.deathMessage = deathMessage;
+     }
+ 
+@@ -53,9 +74,32 @@ public class PlayerDeathEvent extends EntityDeathEvent {
+      *
+      * @return Message to appear to other players on the server.
+      */
++    public @Nullable net.kyori.adventure.text.Component deathMessage() {
++        return this.deathMessage;
++    }
++    // Paper end
++
++    /**
++     * Set the death message that will appear to everyone on the server.
++     *
++     * @param deathMessage Message to appear to other players on the server.
++     * @deprecated in favour of {@link #deathMessage(net.kyori.adventure.text.Component)}
++     */
++    @Deprecated // Paper
++    public void setDeathMessage(@Nullable String deathMessage) {
++        this.deathMessage = deathMessage != null ? net.kyori.adventure.text.serializer.legacy.LegacyComponentSerializer.legacySection().deserialize(deathMessage) : net.kyori.adventure.text.Component.empty(); // Paper
++    }
++
++    /**
++     * Get the death message that will appear to everyone on the server.
++     *
++     * @return Message to appear to other players on the server.
++     * @deprecated in favour of {@link #deathMessage()}
++     */
+     @Nullable
++    @Deprecated // Paper
+     public String getDeathMessage() {
+-        return deathMessage;
++        return net.kyori.adventure.text.serializer.legacy.LegacyComponentSerializer.legacySection().serialize(this.deathMessage); // Paper
```

Comment:
AFAIK Kyori stuff doesn't like nulls. Does this return null upon null input?

Reply:
This will broken, will throw if `deathMessage` is `null`